blockchain/utxoviewpoint.go

```diff
@@ -533,25 +533,25 @@ func (view *UtxoViewpoint) fetchUtxosMain(db database.DB, outpoints map[wire.Out
 // fetchUtxos loads the unspent transaction outputs for the provided set of
 // outputs into the view from the database as needed unless they already exist
 // in the view in which case they are ignored.
-func (view *UtxoViewpoint) fetchUtxos(db database.DB, outpoints map[wire.OutPoint]struct{}) error {
+func (view *UtxoViewpoint) fetchUtxos(db database.DB, outpoints []wire.OutPoint) error {
 	// Nothing to do if there are no requested outputs.
 	if len(outpoints) == 0 {
 		return nil
 	}
 
 	// Filter entries that are already in the view.
-	neededSet := make(map[wire.OutPoint]struct{})
-	for outpoint := range outpoints {
+	needed := make([]wire.OutPoint, 0, len(outpoints))
+	for i := range outpoints {
```

Comment:
nit: could do `for _, outpoint := range outpoints` to minimize diff

Reply:
No, I explicitly asked to use the index based access, because the `append()` would otherwise not work as expected because of the Golang loop variable issue (see my resolved comments above).

Reply:
I don't think that is an issue in this case, since it's a non-ref struct being used?

That being said, better safe than sorry 😀

Reply:
Hmm, you're right, in this case it's not problematic. But we've been burned by this a couple of times in the past, so I basically only use the key/index based access. Also makes it easier to review if you know the loop bug can't bite you.